pktgen_util.h

```diff
@@ -15,6 +18,50 @@ typedef struct rte_mbuf** mbuf_array_t;
 
 struct rte_mbuf tx_mbuf_template[RTE_MAX_LCORE];
 
+static void 
+setup_daemon(void)
+{
+
+    FILE *f;
+    pid_t pid;
+
+    pid = fork();
+    
+    if (pid < 0) {
+        exit(EXIT_FAILURE);
+    }
+    if (pid > 0) {
+        exit(EXIT_SUCCESS);
+    }
+    if (setsid() < 0) {
+        exit(EXIT_FAILURE);
+    }
+    pid = fork();
+    if (pid < 0) {
+        exit(EXIT_FAILURE);
+    }
+    if (pid > 0) {
+        exit(EXIT_SUCCESS);
+    }
+
+    umask(0);   
+
+    f = fopen("./pktgen.pid", "w+");
+    fprintf(f, "%d", getpid());
```

Comment:
in the event that fopen returns NULL or fprintf fails, we should panic() here with something along the lines of "failed to write pid"